### PR TITLE
Validate Profile filters

### LINKED ISSUE #1368: Better validation for filters in portal profile

### LINKED ISSUE #1370: Filter 'Time period' has no validation
You can input the value you want, letter, number anything in time period filter

### LINKED ISSUE #1371: Filter 'URI' has no validation

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Validate the profile filters 
## Impacts

checkup and the Admin GUI Profile filter
## NEWS file entries

New Features
++++++++++++
- Validation of profile filters
## Bug Fixes

Fixes #1371
Fixes #1370
Fixes #1368
## Delete branch after merge

NO

### CODE REVIEW

**Comment by julsemaan** [comment]
@jrouzierinverse + @louismunro 

Is this really still for v6 ?

Would be a bit late to bring in new code.

**Comment by louismunro** [comment]
No. 
Moved to 6.1.

**Comment by jrouzierinverse** [comment]
Please review

File: html/pfappserver/lib/pfappserver/Form/Field/ProfileFilter.pm [hunk 1]
```diff
@@ -74,6 +75,17 @@ sub options_type {
       sort keys %pf::factory::condition::profile::PROFILE_FILTER_TYPE_TO_CONDITION_TYPE;
 }
 
+sub validate {
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Field/ProfileFilter.pm:84]: zi doc

File: lib/pf/pfcmd/checkup.pm [hunk 2]
```diff
@@ -1101,8 +1103,9 @@ sub portal_profiles {
                 if ( $key !~ /$profile_params/ );
             if ($key eq 'filter') {
                 foreach my $filter (@{$data->{filter}}) {
-                    add_problem( $FATAL, "Filter '$filter' is invalid for profile '$portal_profile' please update to newer format 'type:data'" )
-                        unless $filter =~ $pf::factory::condition::profile::PROFILE_FILTER_REGEX;
+                    my ($rc, $message) = $validator->validate($filter);
+                    add_problem( $FATAL, "Filter '$filter' is invalid for profile '$portal_profile': $message ")
+                        unless $rc;
```
Comment (julsemaan) [inline lib/pf/pfcmd/checkup.pm]: Not sure it is really better to put it on a new line below. Can be easy to miss to the untrained eye

File: lib/pf/validation/profile_filters.pm [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+sub validate_switch {
+    my ($self, $type, $value) = @_;
+    if (!exists $pf::SwitchFactory::SwitchConfig{$value} || $value eq 'default' || $value =~ /^group / ) {
+        return ($FALSE, "'$value' is an invalid switch id for filter type $type");
+    }
+    return ($TRUE, undef);
+}
+
+
+=head2 validate_switch
```
Comment (julsemaan) [inline lib/pf/validation/profile_filters.pm]: did a typo here

**Comment by julsemaan** [comment]
Other than the :nail_care: stuff, its good

**Comment by jrouzierinverse** [comment]
💅  is done

**Pull request merged**